yt_dlp/extractor/rokfin.py

```diff
@@ -254,3 +422,91 @@ def _real_extract(self, url):
+    IE_NAME = 'rokfin:search'
+    _SEARCH_KEY = 'rkfnsearch'
+    _NETRC_MACHINE = False
+    _BASE_URL = 'https://rokfin.com'
+    _CACHE_SECTION_NAME = 'rokfin-search-engine-access'
+    _search_engine_access_url = None
+    _search_engine_access_key = None
+
+    def _real_initialize(self):
+        self._search_engine_access_url = self._downloader.cache.load(self._CACHE_SECTION_NAME, 'url')
+        self._search_engine_access_key = self._downloader.cache.load(self._CACHE_SECTION_NAME, 'key')
+        if not self._search_engine_access_url or not self._search_engine_access_key:
+            self._search_engine_access_url, self._search_engine_access_key = self._get_search_engine_access_credentials()
+
+    def _search_results(self, query):
+        def get_video_data(metadata):
+            for search_result in metadata.get('results', []):
+                video_id_ind, video_type = self._TYPES.get(traverse_obj(search_result, ('content_type', 'raw')), (None, None))
+                video_id = traverse_obj(search_result, video_id_ind, expected_type=int_or_none)
+                if not video_id or not video_type:
+                    self.write_debug(msg=f'skipping {search_result}')
+                    continue
+                yield self.url_result(url=f'{self._BASE_URL}/{video_type}/{video_id}')
+        if not query:
+            return
+        query_data = {'query': query, 'page': {'size': 100}}
+        total_pages = None
+        for page_number in itertools.count(1):
+            query_data['page']['current'] = page_number
+            search_results = self._run_search_query(
+                data=query_data,
+                note='Downloading search results page %d%s' % (page_number, format_field(total_pages, template=' of ~%d') if total_pages and total_pages >= page_number else ''),
+                errnote='Unable to download search results page %d%s' % (page_number, format_field(total_pages, template=' of ~%d') if total_pages and total_pages >= page_number else ''))
+            total_pages = traverse_obj(search_results, ('meta', 'page', 'total_pages'), expected_type=int_or_none)
+            yield from get_video_data(search_results)
+            if not search_results.get('results'):
+                return
+
+    def _run_search_query(self, data, note, errnote):
+        data_bytes = json.dumps(data).encode('utf-8')
+        search_results = self._download_json(
+            self._search_engine_access_url, self._SEARCH_KEY, note=note, errnote=errnote, fatal=False,
+            encoding='utf-8', data=data_bytes, headers={'authorization': self._search_engine_access_key})
+        if search_results is not False:
+            return search_results
+        self.write_debug('updating access credentials')
+        self._search_engine_access_url = self._search_engine_access_key = None
+        self._downloader.cache.store(self._CACHE_SECTION_NAME, 'url', None)
+        self._downloader.cache.store(self._CACHE_SECTION_NAME, 'key', None)
```

Comment:
unnecessary

Reply:
In the latest version, these three lines became
```
self._db_url = self._db_access_key = None
self._downloader.cache.store(self.ie_key(), 'auth', None)
```
(Wiping out the invalid URL/key pair from cache). Does this solve the issue?

Reply:
But why do u have to to wipe it from cache? You are immediately replacing these values in the next lines

Reply:
In that case, an error is raised, no?